# DESAFIO 1 ANALISIS ESTADISTICO CON PYTHON-DESAFIO LATA

In [5]:
#Importamos librerias 
import pandas as pd
import numpy as np

In [6]:
# Importamos el archivo y creamos el DataFrame 'df_salarios'
df_salaries = pd.read_csv('ds_salaries.csv')
df_salaries.head()

work_year experience_level employment_type                 job_title  \
0       2023               SE              FT  Principal Data Scientist   
1       2023               MI              CT               ML Engineer   
2       2023               MI              CT               ML Engineer   
3       2023               SE              FT            Data Scientist   
4       2023               SE              FT            Data Scientist   

   salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0   80000             EUR          85847                 ES           100   
1   30000             USD          30000                 US           100   
2   25500             USD          25500                 US           100   
3  175000             USD         175000                 CA           100   
4  120000             USD         120000                 CA           100   

  company_location company_size  
0               ES            L  
1               US            S  
2               US            S  
3               CA            M  
4               CA            M

### 1. El promedio general de los salarios, desviación estándar, quintiles y rango. (hint:
###    escoge la columna adecuada para comparar)

In [24]:
# Determinamos que la columna mas adecuada es 'salary_in_usd', ya que nos permite homologar las distintas monedas en los que viene presentada la informacion

promedio = df_salaries['salary_in_usd'].mean()
desv = df_salaries['salary_in_usd'].std()
quintile1 = df_salaries['salary_in_usd'].quantile(0.20)
quintile2 = df_salaries['salary_in_usd'].quantile(0.40)
quintile3 = df_salaries['salary_in_usd'].quantile(0.60)
quintile4 = df_salaries['salary_in_usd'].quantile(0.80)
rango = df_salaries['salary_in_usd'].max() - df_salaries['salary_in_usd'].min()

print(f'El salario promedio es US${promedio}')
print(f'El desviacion estandar del salario es US${desv}')
print(f'El primer quintil esta en US${quintile1}')
print(f'El primer quintil esta en US${quintile2}')
print(f'El primer quintil esta en US${quintile3}')
print(f'El primer quintil esta en US${quintile4}')
print(f'El rango es US${rango}')

El salario promedio es US$137570.38988015978
El desviacion estandar del salario es US$63055.625278224084
El primer quintil esta en US$83972.8
El primer quintil esta en US$120000.0
El primer quintil esta en US$149850.0
El primer quintil esta en US$185900.0
El rango es US$444868


In [9]:
df_salaries.head()

work_year experience_level employment_type                 job_title  \
0       2023               SE              FT  Principal Data Scientist   
1       2023               MI              CT               ML Engineer   
2       2023               MI              CT               ML Engineer   
3       2023               SE              FT            Data Scientist   
4       2023               SE              FT            Data Scientist   

   salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0   80000             EUR          85847                 ES           100   
1   30000             USD          30000                 US           100   
2   25500             USD          25500                 US           100   
3  175000             USD         175000                 CA           100   
4  120000             USD         120000                 CA           100   

  company_location company_size  
0               ES            L  
1               US            S  
2               US            S  
3               CA            M  
4               CA            M

### 2. Una comparación entre los salarios agrupados por 3 diferentes categorías a tu
###    elección. ¿Para cuál de ellas las medidas de tendencia central son más
###    representativas? ¿Para cuál son menos representativas? Explica.

In [44]:
pd.pivot_table(df_salaries,
               index='job_title',
               values='salary_in_usd',
               aggfunc=['mean', 'std', 'max', 'min'])

mean            std           max  \
                               salary_in_usd  salary_in_usd salary_in_usd   
job_title                                                                   
3D Computer Vision Researcher   21352.250000   20046.470685         50000   
AI Developer                   136666.090909   95973.553918        300000   
AI Programmer                   55000.000000   21213.203436         70000   
AI Scientist                   110120.875000  111733.541359        423834   
Analytics Engineer             152368.631068   53838.467623        289800   
...                                      ...            ...           ...   
Research Engineer              163108.378378   55055.424934        293000   
Research Scientist             161214.195122   74944.431627        450000   
Software Data Engineer          62510.000000   17691.811665         75020   
Staff Data Analyst              15000.000000            NaN         15000   
Staff Data Scientist           105000.000000            NaN        105000   

                                        min  
                              salary_in_usd  
job_title                                    
3D Computer Vision Researcher          5409  
AI Developer                           6304  
AI Programmer                         40000  
AI Scientist                          12000  
Analytics Engineer                     7500  
...                                     ...  
Research Engineer                     60000  
Research Scientist                    23000  
Software Data Engineer                50000  
Staff Data Analyst                    15000  
Staff Data Scientist                 105000  

[93 rows x 4 columns]

In [26]:
pd.pivot_table(df_salaries,
               index='experience_level',
               values='salary_in_usd',
               aggfunc=['mean', 'std', 'max', 'min'])

mean           std           max           min
                  salary_in_usd salary_in_usd salary_in_usd salary_in_usd
experience_level                                                         
EN                 78546.284375  52225.424309        300000          5409
EX                194930.929825  70661.929661        416000         15000
MI                104525.939130  54387.685128        450000          5132
SE                153051.071542  56896.263954        423834          8000

In [32]:
pd.pivot_table(df_salaries,
               index='employment_type',
               values='salary_in_usd',
               aggfunc=['mean', 'std', 'max', 'min'])

mean            std           max           min
                 salary_in_usd  salary_in_usd salary_in_usd salary_in_usd
employment_type                                                          
CT               113446.900000  130176.746842        416000          7500
FL                51807.800000   29458.879336        100000         12000
FT               138314.199570   62452.177613        450000          5132
PT                39533.705882   38312.145181        125404          5409

### 3. ¿Qué cargos reciben mejores sueldos en las empresas con sede en Estados Unidos?

In [37]:
# Creamos un DataFrame solo de las empresas con sede en Estados Unidos
df_usa = df_salaries[df_salaries['company_location'] == 'US']
df_usa

work_year experience_level employment_type                 job_title  \
1          2023               MI              CT               ML Engineer   
2          2023               MI              CT               ML Engineer   
5          2023               SE              FT         Applied Scientist   
6          2023               SE              FT         Applied Scientist   
9          2023               SE              FT            Data Scientist   
...         ...              ...             ...                       ...   
3749       2021               SE              FT           Data Specialist   
3750       2020               SE              FT            Data Scientist   
3751       2021               MI              FT  Principal Data Scientist   
3752       2020               EN              FT            Data Scientist   
3753       2020               EN              CT     Business Data Analyst   

      salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
1      30000             USD          30000                 US           100   
2      25500             USD          25500                 US           100   
5     222200             USD         222200                 US             0   
6     136000             USD         136000                 US             0   
9     147100             USD         147100                 US             0   
...      ...             ...            ...                ...           ...   
3749  165000             USD         165000                 US           100   
3750  412000             USD         412000                 US           100   
3751  151000             USD         151000                 US           100   
3752  105000             USD         105000                 US           100   
3753  100000             USD         100000                 US           100   

     company_location company_size  
1                  US            S  
2                  US            S  
5                  US            L  
6                  US            L  
9                  US            M  
...               ...          ...  
3749               US            L  
3750               US            L  
3751               US            L  
3752               US            S  
3753               US            L  

[3040 rows x 11 columns]

In [43]:
high_jobs = df_usa.groupby('job_title')['salary_in_usd'].mean()
high_jobs.sort_values(ascending=False)

job_title
Data Analytics Lead          405000.000000
Data Science Tech Lead       375000.000000
Director of Data Science     294375.000000
Principal Data Scientist     255500.000000
Cloud Data Architect         250000.000000
                                 ...      
Data Quality Analyst          82863.333333
BI Data Analyst               82709.833333
Data Analytics Consultant     81500.000000
Big Data Engineer             70000.000000
BI Data Engineer              60000.000000
Name: salary_in_usd, Length: 70, dtype: float64

### Algunas de los comentarios que podemos hacer al respecto tenemos los siguientes:
- Altos Ingresos para Liderazgo en Análisis de Datos: El puesto de "Data Analytics Lead" tiene el salario promedio más alto, con 405,000 USD al año, destacando la alta demanda y el valor significativo de los líderes en análisis de datos.

- Competencia en Liderazgo Técnico de Ciencia de Datos: El "Data Science Tech Lead" sigue de cerca, con un salario promedio de 375,000 USD, lo que subraya la importancia y el alto valor de los líderes técnicos especializados en ciencia de datos.

- Variabilidad en Roles de Ciencia de Datos y Arquitectura de Datos: Otros puestos bien remunerados incluyen "Director of Data Science" (294,375 USD), "Principal Data Scientist" (255,500 USD) y "Cloud Data Architect" (250,000 USD), mostrando que los roles directivos y arquitectónicos en ciencia de datos también son altamente valorados.

- Diversidad en Salarios de Ingenieros y Científicos de Datos Aplicados: Roles como "Applied Data Scientist" (238,000 USD) y "Machine Learning Software Engineer" (217,400 USD) tienen salarios competitivos, indicando una demanda considerable tanto para científicos de datos aplicados como para ingenieros de software especializados en aprendizaje automático.

Este resumen refleja la alta remuneración y la importancia de los roles de liderazgo y especialización en el área de TI, especialmente en ciencia de datos y análisis. 

Sin embargo al revisar datos en internet se puede apreciar que para un 'Data Analytics Lead' los sueldos promedios anuales rondan entre los 112,000 USD y los 152,000 USD, por lo que recomendaria revisar la fuente de los datos o como fueron recopilados (fuente: https://www.glassdoor.com.mx/Sueldos/estados-unidos-lead-data-science-analyst-sueldo-SRCH_IL.0,14_IN1_KO15,40.htm)